### ISSUE TITLE
Order of Envoy sidecar being injected into the containers list.

### ISSUE THREAD

**Issue by allenlsy** [openservicemesh/osm#4586]
**Bug description**:

OSM injector currently **appends** Envoy sidecar into the pod containers list. It is very likely that Envoy will be the last container in the list. The order of the container matters in some cases. Kubelet creates containers sequentially (and non-blocking) according to the list. We have a use case that want to ensure Envoy is completely ready before starting application container, to avoid race condition between Envoy and application. If application is ready before Envoy, some requests will fail.

https://github.com/openservicemesh/osm/blob/6acc95394e087f87d809bf8f821490d49fb6bead/pkg/injector/patch.go#L97

The suggestion is to prepend Envoy container to this list.

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**:

Envoy container should be the first of the containers list

**Steps to reproduce the bug (as precisely as possible)**:

In OSM demo, check the containers list of any application, E.g. bookbuyer.

![image](https://user-images.githubusercontent.com/872876/157727332-c1c528bd-529f-437e-8a28-f96eb7fb0c0f.png)

**How was OSM installed?**:

OSM is install with v1.0 release. 

**Anything else we need to know?**:

**Bug report archive**:

**Environment**:
- OSM version (use `osm version`): 1.0
- Kubernetes version (use `kubectl version`): 1.21
- Size of cluster (number of worker nodes in the cluster): 2
- Others:

**Comment by shashankram** [comment]
Ordering does not ensure app will startup correctly if it depends on network connectivity. In fact, it also needs Envoy configuration to be correctly configured by the controller before the app can be ready to send traffic over the network.

I am closing this as a duplicate of https://github.com/openservicemesh/osm/issues/2316 which pertains to resolving app startup issues due to Envoy. @allenlsy, refer to 3. in https://github.com/openservicemesh/osm/issues/2316#issue-789249802 which covers the ordering problem.
